task/sql-tasks.js

```diff
@@ -281,7 +460,26 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+        SELECT DISTINCT
+            CompanyName,
+            Products.ProductName,
+            OrderDetails.UnitPrice AS "PricePerItem"
+        FROM Customers
+         INNER JOIN Orders on Customers.CustomerID = Orders.CustomerID
+         INNER JOIN OrderDetails on Orders.OrderID = OrderDetails.OrderID
+         INNER JOIN Products on  Products.ProductID = OrderDetails.ProductID
+        WHERE OrderDetails.UnitPrice = (
+            SELECT
+                MAX(ordrDtls.UnitPrice)
+            FROM Customers cstmrs
+             INNER JOIN Orders ordrs on cstmrs.CustomerID = ordrs.CustomerID
+             INNER JOIN OrderDetails ordrDtls on ordrs.OrderID = ordrDtls.OrderID
+            WHERE Customers.CompanyName = cstmrs.CompanyName

```

Comment:
companyname doesn't uniquely identify customer according to the schema. Also you don't have index for this field so it is also bad for performance.